flox-bash/lib/commands/general.sh

```diff
@@ -165,17 +177,25 @@ function floxSearch() {
 			showDetail="true"
 			shift
 			;;
+		*@*)  # Semver
+			semver=:
+			semverRange="${1##*@}"
+			_pkg="${1%@*}"
+			shift
+			set -- "$_pkg" "$@"
+			unset _pkg
+		    ;;
 		*)
-			if [ "$subcommand" = "packages" ]; then
+			if [[ "${subcommand:-}" = "packages" ]]; then
```

Comment:
What is this change doing?

Reply:
1. If the variable is unset `bash` will not throw an error here. `"${var:-}"` explicitly states "this variable is allowed to be unset".

2. The `[` -> `[[` change is an optimization which uses the builtin `bash` command `[[` instead of the system's `[` ( `expr` ) executable. It's small but given how frequently scripts use that executable it makes an impact.